Description: Serialize via dill
Author: Sandro Tosi <morph@debian.org>
Forwarded: no
Last-Update: 2024-01-15

--- a/src/pydot/core.py
+++ b/src/pydot/core.py
@@ -432,7 +432,11 @@
         return dict
 
     def __setstate__(self, state):
-        self.obj_dict = state
+        self.obj_dict = state['obj_dict']
+
+    def __reduce__(self):
+        state = self.__dict__.copy()
+        return (Common, (), state, )
 
     def __get_attribute__(self, attr):
         """Look for default attributes for this node"""
@@ -1594,7 +1598,11 @@
         return dict
 
     def __setstate__(self, state):
-        self.obj_dict = state
+        self.obj_dict = state['obj_dict']
+
+    def __reduce__(self):
+        state = self.__dict__.copy()
+        return (Dot, (), state, )
 
     def set_shape_files(self, file_paths):
         """Add the paths of the required image files.
